code/game/mecha/equipment/weapons/weapons.dm

```diff
@@ -187,29 +223,42 @@
 	equip_cooldown = 20
 	projectile = /obj/item/projectile/bullet/midbullet
 	fire_sound = 'sound/weapons/shotgun.ogg'
-	projectiles = 40
+	max_projectiles = 40
 	projectile_energy_cost = 25
 	var/projectiles_per_shot = 4
-	var/deviation = 0.7
+//	var/deviation = 0.7  //the shots were perfectly accurate no matter what this was set to
 
 /obj/item/mecha_parts/mecha_equipment/weapon/ballistic/scattershot/action(atom/target)
 	if(!action_checks(target))
 		return
+	var/originaltarget = target
 	var/turf/curloc = get_turf(chassis)
 	var/turf/targloc = get_turf(target)
 	if(!curloc || !targloc)
 		return
-	var/target_x = targloc.x
-	var/target_y = targloc.y
-	var/target_z = targloc.z
-	targloc = null
+//	var/target_x = targloc.x
+//	var/target_y = targloc.y
+//	var/target_z = targloc.z
+//	targloc = null
 	for(var/i=1 to min(projectiles, projectiles_per_shot))
-		targloc = locate(target_x+GaussRandRound(deviation,1),target_y+GaussRandRound(deviation,1),target_z)
+//		targloc = locate(target_x+GaussRandRound(deviation,1),target_y+GaussRandRound(deviation,1),target_z)
+		if(damaged)
+			var/list/turf/shot_spread = list()
+			for(var/turf/T in range(originaltarget, min(2, max(0, get_dist(chassis, originaltarget)-1))))
```

Comment:
Again, `trange()`.

On another note: this seems copy pasted from the above code block, you should probably try to remove some copy paste.